packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx

```diff
@@ -217,147 +214,151 @@ const DataSpaceQueryBuilderSetupPanelContent = observer(
-                placeholder="Choose an execution context..."
-                options={executionContextOptions}
-                disabled={
-                  executionContextOptions.length < 1 ||
-                  (executionContextOptions.length === 1 &&
-                    Boolean(selectedExecutionContextOption))
-                }
-                onChange={onExecutionContextOptionChange}
-                value={selectedExecutionContextOption}
+                placeholder="Choose a runtime..."
+                noMatchMessage="No compatible runtime found for specified execution context"
+                options={runtimeOptions}
+                onChange={changeRuntime}
+                value={selectedRuntimeOption}
                 darkMode={
                   !applicationStore.layoutService
                     .TEMPORARY__isLightColorThemeEnabled
                 }
+                filterOption={runtimeFilterOption}
+                formatOptionLabel={getRuntimeOptionFormatter({
+                  darkMode:
+                    !applicationStore.layoutService
+                      .TEMPORARY__isLightColorThemeEnabled,
+                })}
               />
             </div>
-            {queryBuilderState.showRuntimeSelector && (
-              <div className="query-builder__setup__config-group__item">
-                <div
-                  className="btn--sm query-builder__setup__config-group__item__label"
-                  title="runtime"
-                >
-                  <PURE_RuntimeIcon />
-                </div>
-                <CustomSelectorInput
-                  className="panel__content__form__section__dropdown query-builder__setup__config-group__item__selector"
-                  placeholder="Choose a runtime..."
-                  noMatchMessage="No compatible runtime found for specified execution context"
-                  options={runtimeOptions}
-                  onChange={changeRuntime}
-                  value={selectedRuntimeOption}
-                  darkMode={
-                    !applicationStore.layoutService
-                      .TEMPORARY__isLightColorThemeEnabled
-                  }
-                  filterOption={runtimeFilterOption}
-                  formatOptionLabel={getRuntimeOptionFormatter({
-                    darkMode:
-                      !applicationStore.layoutService
-                        .TEMPORARY__isLightColorThemeEnabled,
-                  })}
-                />
-              </div>
-            )}
+          )}
+          <div className="query-builder__setup__config-group__item">
+            <QueryBuilderClassSelector
+              queryBuilderState={queryBuilderState}
+              classes={classes}
+              onClassChange={queryBuilderState.onClassChange}
+              noMatchMessage="No compatible class found for specified execution context"
```

Comment:
do we need to rename from `class` to `entity` for consistency

Reply:
Done